### PR TITLE
Prompt user to save as if save fails

### LINKED ISSUE #7708: Unsaved changes lost if permission denied when saving on exit
If I open a file that I do not have permission to save and try to save it I get a popup telling me I lack permissions, e.g. `Unable to save file: permission denied '/etc/hosts'`. This seems reasonable (although I wish it would ask me for my sudo password instead).
However, if I modify it and then close it, Atom will ask me if I want to save it, and when I click "yes", it closes the file and gives me that same popup that it failed to save. This means my work is lost. I don't think I have used an editor before that doesn't keep the file open if it fails to save and I believe this should be considered a bug, right?

I am on Chakra linux running 1.0.0 via http://chakraos.org/ccr/packages.php?ID=7252

### PR DESCRIPTION
**Pull request by kuychaco**
Thanks @einarmagnus for adding this feature :sparkles:! I resolved the merge conflicts and updated code to use the current APIs.

Closes https://github.com/atom/atom/pull/7713
Closes https://github.com/atom/atom/issues/7708

### CODE REVIEW

File: src/pane.coffee [hunk 1]
```diff
@@ -577,13 +577,26 @@ class Pane extends Model
     else
       return true
 
+    saveError = (error) =>
+      if error
+        chosen = @applicationDelegate.confirm
```
Comment (lee-dohm) [inline src/pane.coffee]: I'd like to see this unified with the nearly duplicate code starting on line 593.

File: src/pane.coffee [hunk 2]
```diff
@@ -577,13 +577,26 @@ class Pane extends Model
...
+          message: @getMessageForErrorCode(error)
```
Comment (lee-dohm) [inline src/pane.coffee]: The other version starting on line 593 definitely identifies the file that has changes and will be saved. This message doesn't seem to do so. Can that be added?

**Comment by amytruong** [comment]
**SCENARIOS TESTED**
 **Mac 10.10**
Close out edited read-only file - **PASS**
1. Make file read-only
2. Edit file
3. Close out file 
4. Save dialog box displays. Click Save
5. Atom Helper wants to make changes box displays.
6. Click Cancel (instead of putting in password)
**Expected Results:** Permission denied message displayed with **Save As** dialog box displaying
**Actual Results:** Permission denied message displayed with dialog box displaying. 

The save-as dialog box doesn't display, if I click Cancel after entering in my password, just the error message if you click Save/CMD-S - **FAIL** _(Not sure if this scenario falls in the scope of the fix but tried it out anyway just in case)_
1. Make file read-only
2. Make changes in Atom
3. Click Save/cmd-S
4. Atom Helper wants to make changes box displays
5. Click Cancel (instead of putting in password)
**Expected Results:** Permission denied message displayed and  **Save As** dialog box also displayed
**Actual Results:** Permission denied message displayed **_without_** Save As dialog box 
The message is only displayed. :cry:
screen shot 2016-04-18 at 12 08 45 pm

 **Windows 8.1**
Close out edited read-only file **PASS**
1. Make file read-only
2. Edit file
3. Close out file 
4. Save dialog box displays. Click Save
**Expected Results:** Permission denied message displayed with **Save As** dialog box displaying
**Actual Results:** Permission denied message displayed with **Save As** dialog box displaying

The save-as dialog box doesn't display if I click Cancel after entering in my password, just the error message if you click Save/CMD-S  _(Not sure if this scenario falls in the scope of the fix but tried it out anyway just in case)_
1. Make file read-only
2. Make changes in Atom
3. Click Save/cmd-S
4. Atom Helper wants to make changes box displays.
5. Click Cancel (instead of putting in password)
**Expected Results:** Permission denied message displayed with **Save As** dialog box displaying
**Actual Results:** Permission denied message displayed **_without_** Save As dialog box 
The message is only displayed. :cry:
screen shot 2016-04-18 at 12 50 48 pm

**Comment by lee-dohm** [comment]
I think this is a definite improvement over the current behavior and I'm not sure that I was able to find the same issues that @amytruong did. I'm going to merge this for now so we don't hold up the improvement on the edge case. Especially because the edge case doesn't enable people to lose data like the old behavior (the tab stays open in all cases from what I can see).

**Pull request merged**